kmap.lisp

```diff
@@ -117,8 +117,8 @@ for passing as the last argument to (apply #'make-key ...)"
   (unless (evenp end)
     (signal 'kbd-parse-error :string mods))
   (apply #'nconc (loop for i from 0 below end by 2
-                       if (char/= (char mods (1+ i)) #\-)
-                       do (signal 'kbd-parse)
+                       when (char/= (char mods (1+ i)) #\-)
+                         do (signal 'kbd-parse-error)
```

Comment:
Signalling `kbd-parse-error` without a `:string` argument leaves the slot in the error unbound, which will trigger an error when printing the condition. This should probably be the same as 3 lines up:

```lisp
(signal 'kbd-parse-error :string mods)
```

A example of when this goes wrong is evaluating `(kbd "Shift_L-Up")` through the `eval` command. The enclosing `handler-case` catches the error and tries to print it.

Thinking about this some more, I'm a bit confused why the code uses `signal` on an `error`. If anything in the callstack is set up to catch `error` or one of its superclasses, the error will be caught. With StumpWM's top-level condition handler, this basically means a call like `(kbd "Shift_L-Up")` will never really succeed inside the StumpWM event loop. When running in the REPL, however, it will appear to run just fine.